controllers/validator.go

```diff
@@ -63,30 +60,14 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
 		}
 	}
 
-	// check if the VSM plugin is specified or not
-	VSMPluginPresent := false
+	// check for VSM/Volsync DataMover (OADP 1.2 or below) syntax
 	for _, plugin := range dpa.Spec.Configuration.Velero.DefaultPlugins {
 		if plugin == oadpv1alpha1.DefaultPluginVSM {
-			VSMPluginPresent = true
+			return false, errors.New("Delete vsm from spec.configuration.velero.defaultPlugins and dataMover object from spec.features, they are not used in OADP 1.3")
 		}
 	}
-
-	if r.checkIfDataMoverIsEnabled(&dpa) {
-		// parse for timeout if specified and see if there are no errors
-		if len(dpa.Spec.Features.DataMover.Timeout) > 0 {
-			_, err := time.ParseDuration(dpa.Spec.Features.DataMover.Timeout)
-			if err != nil {
-				return false, err
-			}
-		}
-
-		if !VSMPluginPresent {
-			return false, errors.New("datamover is enabled, specify vsm as a default plugin")
-		}
-	}
-
-	if !r.checkIfDataMoverIsEnabled(&dpa) && VSMPluginPresent {
-		return false, errors.New("datamover is disabled, remove vsm as a default plugin")
+	if dpa.Spec.Features != nil && dpa.Spec.Features.DataMover != nil {
```

Comment:
This is not needed if we directly remover `Datamover` from `Features` , right ?

Reply:
yes, but as we discussed, it better to left it there and tell user to remove it during DPA reconcile, otherwise, upgrade can break and get stuck 

reason https://github.com/openshift/oadp-operator/pull/1171#discussion_r1362431555